frontend/packages/ux-editor/src/components/config/Expressions/NewExpressionButton/NewExpressionButton.tsx

```diff
@@ -12,43 +12,58 @@ export const NewExpressionButton = () => {
   const t = useText();
   const { formItem } = useFormItemContext();
 
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+
   const remainingProperties = getUndefinedExpressionProperties(formItem);
   const areThereRemainingProperties = remainingProperties.length > 0;
 
   const title = areThereRemainingProperties
     ? t('right_menu.expressions_add')
     : t('right_menu.expressions_expressions_limit_reached_alert');
 
-  return (
-    <StudioDropdownMenu
-      anchorButtonProps={{
-        children: t('right_menu.expressions_add'),
-        color: 'first',
-        disabled: !areThereRemainingProperties,
-        icon: <PlusIcon />,
-        title,
-        variant: 'secondary',
-      }}
-      size='small'
-    >
-      {remainingProperties.map((property) => (
-        <MenuItem property={property} key={JSON.stringify(property)} />
-      ))}
-    </StudioDropdownMenu>
-  );
-};
+  const handleMenuToggle = () => setIsMenuOpen((prev) => !prev);
 
-const MenuItem = ({ property }: { property: FormItemProperty }) => {
-  const { formItem, handleUpdate, debounceSave } = useFormItemContext();
-  const text = useAddExpressionText(property);
+  useEffect(() => {
+    if (isMenuOpen) document.addEventListener('click', handleMenuToggle);
+    return document.removeEventListener('click', handleMenuToggle);
+  }, [isMenuOpen]);
 
-  const handleAddExpression = async () => {
-    const newFormItem = addExpressionToFormItem(formItem, property);
-    handleUpdate(newFormItem);
-    await debounceSave(); // Todo: handleSave does not work here. Will probably be fixed by https://github.com/Altinn/altinn-studio/issues/12383.
+  const MenuItem = ({ property }: { property: FormItemProperty }) => {
+    const { handleUpdate, debounceSave } = useFormItemContext();
+    const text = useAddExpressionText(property);
+
+    const handleAddExpression = async () => {
+      setIsMenuOpen(true);
+      const newFormItem = addExpressionToFormItem(formItem, property);
+      handleUpdate(newFormItem);
+      await debounceSave(); // Todo: handleSave does not work here. Will probably be fixed by https://github.com/Altinn/altinn-studio/issues/12383.
+      setIsMenuOpen(false);
+    };
+
+    return <DropdownMenu.Item onClick={handleAddExpression}>{text}</DropdownMenu.Item>;
```

Comment:
I think the actual issue here is that our component here renders the DropdownMenu.Item from DS directly and not from Studio, so the `setOpen` callback is lost 🫣 

I believe that all this issue needs is to replace this one with:
```suggestion
    return <StudioDropdownMenu.Item onClick={handleAddExpression}>{text}</StudioDropdownMenu.Item>;
```

Reply:
Amazing❤️, It works fine, I revert all changes